cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -607,6 +586,45 @@ public void onBecomeFollowerFromOffline(Message message, NotificationContext con
       }
     }
 
+    // rebuild the replica if there is at least one live replica to rebuild from AND it meets either of TWO conditions:
+    // Case 1. DB local meta is different from upstream
+    // Case 2. DB local meta is the same as upstream, but latest local data update is too old (WAL TTL expired).
+    //         However, skip rebuild if sequence number is the same as upstream (nothing to rebuild from).
```

Comment:
could we re-phrase to (I think, use AND or OR is not very easy in our case since we are mixing "AND OR" in our conditions):
We have following conditions to "must rebuild" and "can skip":
"must rebuild" (if either is true):
- local meta is different from upstream
"can skip"
- WAL's latest update timestamp is still fairly new, it could still possible get all the latest data from upstream's WAL via replication
- local seqNo is same as upstream

Reply:
my intention here is not to mix AND/OR, but to ONLY state when a rebuild is necessary. Hence the two cases. The second case is a little nuanced but hopefully should be clear enough with the update. 

I got confused on your latest comment, since "must rebuild" and "can skip" means a reader will still need to figure out when a rebuild is needed. we should avoid vague messages like "can skip", imho.

Reply:
I think the "however" part is still confusing, since it is another exception inside the *Case2*. 
I can't think of a better rephrase now, and current change is at least better than before, thus, r+.